File: online-store-vite-react/src/entities/Product.ts
import { IComment } from "./Comment";
import { ICategory } from "./Category.ts";

export interface IProduct {
  id: number;
  title: string;
  price: number;
  rating: number;
  images: GetProductImageDto[];
  description: string;
  categoryId: number;
  discount?: number;
}

export interface GetProductImageDto {
  id: number;
  image: string;
  productId: number;
}

export interface IProductUpdate {
  title: string;
  description: string;
  price: number;
  rating: number;
  images: File[];
  categoryId: number;
  discount?: number;
  deliveryKit: string;
  isStock: boolean;
}
export interface IProductCreate {
  title: string;
  description: string;
  price: number;
  rating: number;
  images: File[];
  categoryId: number;
  discount?: number;
  deliveryKit: string;
  isStock: boolean;
}

export interface IProductPreview {
  id: number;
  title: string;
  price: number;
  image: string;
  discount: number;
  isFavorite: boolean | null;
  category?: ICategory;
}

export interface IProductDetail {
  id: number;
  title: string;
  price: number;
  images: IProductImage[];
  description: string;
  rating: number;
  comments: IComment[];
  deliveryKit: string;
  discount: number;
  category?: ICategory;
  isStock: boolean;
  isFavorite: boolean | null;
}
export interface IProductList<T> {
  items: T[];
}

export interface IProductImage {
  id: number;
  image: string;
  productId: number;
}

File: online-store-web-api/Core/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities
{
    [Table("Products")]
    public class Product : BaseEntity<int>
    {
        public string? Title { get; set; }
        public decimal Price { get; set; }
        public string? Description { get; set; }
        public string DeliveryKit {  get; set; }
        public ICollection<ProductImage>? Images { get; set; }
        public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        public float Rating { get; set; }
        public int CategoryId { get; set; }
        public bool isStock { get; set; } 
        public Category? Category { get; set; }
        public virtual ICollection<FavoriteProduct> FavoriteProducts { get; set; } = new List<FavoriteProduct>();
        public virtual ICollection<ViewedProduct> ViewedProducts { get; set; } = new List<ViewedProduct>();
        public decimal? Discount { get; set; }

    }
}
